### ISSUE TITLE
repochecker: don't post invalid utf8

### ISSUE THREAD

**Issue by lnussel** [openSUSE/osc-plugin-factory#1058]
Today we had a problem with the source checker that lead repochecker to wanting to decline requests. It failed doing that as it apparently tried to post comments that contained invalid utf8. repochecker should make sure to only post valid strings.

**Comment by jberry-suse** [comment]
On which requests did it fail?

**Comment by jberry-suse** [comment]
Presumably this https://bugzilla.opensuse.org/show_bug.cgi?id=1053969.

**Comment by DimStar77** [comment]
On Wed, 2017-08-16 at 06:42 -0700, Jimmy Berry wrote:
> Presumably this https://bugzilla.opensuse.org/show_bug.cgi?id=1053969
> 

Yes, that's the reason why it wanted to decline submissions - yet,
there is still the issue that the bot crashed on actually performing
the decline task

cheers,
Dominique

**Comment by jberry-suse** [comment]
Yeah, it is better if I can reproduce and ensure fix at least solve the case that was the reason for this issue.

**Comment by lnussel** [comment]
https://build.opensuse.org/request/show/517144 was the one I was looking at

**Comment by jberry-suse** [comment]
Just to clarify talking about review message not a comment.

**Comment by jberry-suse** [comment]
Need to test this, but #1089 may just fix since it does an explicit encode, but not really sure what was causing the issue since I haven't attempted to reproduced.

**Comment by jberry-suse** [comment]
Since the primary issue was broken underlying tool which has since been fixed and the comment code has truncation which is the only time I've seen OBS complain about invalid utf8 perhaps just apply the truncation code to review message, but really this should be reworked to just include the output in comment since it is a lot more readable and consistent with current bot behavior.
